src/System.Net.Security/src/System/Net/Security/SslStream.cs

```diff
@@ -452,7 +452,7 @@ public Task AuthenticateAsServerAsync(SslServerAuthenticationOptions sslServerAu
                 this);
         #endregion
 
-        public override bool IsAuthenticated => _context != null && _context.IsValidContext && _exception == null && _handshakeCompleted;
+        public override bool IsAuthenticated => _exception == null && _handshakeCompleted && (_context?.IsValidContext ?? false);
```

Comment:
Why reorder this?

Reply:
It was initially to tailcall into `IsValidContext->_securityContext.IsInvalid` when I wasn't sure I could get it to devirtualize.

Can revert if preferred?

Reply:
If there's a benefit, add a comment for how someone can make sure they don't undo it.  If there's no benefit, then this file should be completely reverted.

Reply:
Removed; with the devirtualization it no longer provides benefit, it was a step on the way